src/styles/main.scss

```
@@ -1,3 +1,82 @@
+$day-size: 100px;
+$day-color: #eee;
+$calendar-padding: 10px;
+$day-gap: 1px;
+
+@mixin sqare($size) {
+  width: $size;
+  height: $size;
+  background-color: $day-color;
+}
+
 body {
   margin: 0;
 }
+
+.container {
+  padding: 100px;
+  margin: 0 auto;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  padding: $calendar-padding;
+  margin: 0 auto;
+  box-sizing: border-box;
+  width: 6 * $day-gap + 7 * $day-size + 2 * $calendar-padding;
+  transition-duration: 0.5s;
+
+  &-day {
+    @include sqare($day-size);
+    box-sizing: border-box;
+    border: 1px solid black;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &-day:hover {
+    background-color: pink;
+    transform: translateY(-20px);
+    transition-duration: 0.5s;
+    cursor: pointer;
+  }
+
+  @for $index from 1 through 31 {
+    &-day:nth-child(#{$index}):before {
+      font-family: "Arial", sans-serif;
+      color: black;
+      margin: 0 auto;
+      font-size: 30px;
+      line-height: 30px;
+      content: "#{$index}";
+    }
+  }
+
+  $day-names:
+    "mon" 0,
+    "tue" $day-size + $day-gap,
+    "wed" 2 * ($day-size + $day-gap),
+    "thu" 3 * ($day-size + $day-gap),
+    "fri" 4 * ($day-size + $day-gap),
+    "sat" 5 * ($day-size + $day-gap),
+    "sun" 6 * ($day-size + $day-gap);
+
+  @each $name, $size in $day-names {
+    &-day__start-day--#{$name} {
+      margin-left: #{$size};
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &__month-length--#{$index} {
+      .calendar-day:nth-child(n + #{$index}) {
```

now I need to change the modifier to `calendar calendar__month-length--32` to see all the 31 days on the page

```suggestion
      .calendar-day:nth-child(n + #{$index+1}) {
```